app/assets/javascripts/application.js

```diff
@@ -158,6 +158,68 @@ $(document).on('hidden.bs.modal', function(e) {
     OPEN_MODAL = '';
 });
 
+/**
+ * Logs properties about rendered Study Overview plot to Mixpanel via Bard.
+ *
+ * See overview diagram at:
+ * https://app.lucidchart.com/invitations/accept/3b086049-89bc-4db2-96c3-815ccc438bd5
+ *
+ * @param {String} plotType Type of plot, e.g. scatter, violin, box, dot
+ */
+function logPlot(plotType) {
+
+  var endTime = performance.now();
+  var startTimeFrontend = window.SCP.perfTimeStartPlotRender;
+
+  var startTime;
+  if (typeof window.SCP.perfTimeStartPlotTrigger === 'undefined') {
+    // Plot trigger is page navigation, e.g. for "Clusters" scatter plot
+    // startTime = window.performance.timing.navigationStart;
+    //
+    // In other words, for plots that are triggered by merely loading the page,
+    // we consider the "start time" to be the "navigation start" -- e.g.
+    // when they clicked the link to get to this page.  Clicking the link
+    // (or otherwise starting navigation) from that previous page was the
+    // *user action* that triggered this plot.  It most accurately reflects
+    // how long the user waits between doing something and seeing something
+    // they can meaningfully interact with.
+    const perfData = window.performance.timing;
+    const pageLoadTime = perfData.loadEventEnd - perfData.navigationStart;
+
+    endTime += pageLoadTime;
+    startTimeFrontend += pageLoadTime;
+    startTime = pageLoadTime;
+
+  } else {
+    startTime = window.SCP.perfTimeStartPlotTrigger;
+  }
+
+  // TODO (SCP-2736):
+  // * Better delineate backend time w.r.t. time to first byte vs. time-to-last-byte
+  // * Apply concept from legacy Rails plot:* timings to comparable React components
+  var perfTimeJourney = Math.round(endTime - startTime);
+  var perfTime = Math.round(endTime - startTimeFrontend);
+  var perfTimeBackend = perfTimeJourney - perfTime;
+
+  var logProps = {
+    'perfTime': perfTime,
+    'perfTime:journey': perfTimeJourney,
+    'perfTime:backend': perfTimeBackend,
+    currentTab: $('#view-tabs .study-nav.active').text().trim().toLowerCase(),
+    genes: $('#search_genes').val().split(' '),
```

Comment:
This line makes me a little nervous since it will error if the search_genes field isn't on the page yet/anymore, which restricts the places we can use this function.  So maybe `$('#search_genes').val() ? $('#search_genes').val().split(' ') : ''`

Reply:
Fixed in a2f157e.  Using this function beyond Study Overview is currently out-of-scope.  To make that part of the docstring clearer, I've added a TODO right nearby to generalize how we get these props.

Intended use aside, you're right that something was off.  The standard way to get the Mixpanel `genes` prop is to use the just-introduced `formatTerms` function.  That includes refined handling for your suggestion.